### ISSUE TITLE
CA1859 ignores switch expressions

### ISSUE THREAD

**Issue by marek-safar** [dotnet/roslyn-analyzers#6452]
### Analyzer

CA1859

### Analyzer source

https://github.com/dotnet/runtime/blob/8798c0459a36463bf3355f1059ad97fdd890c99e/eng/Versions.props#L40

### Describe the bug

It seems like the analyzer ignores switch expressions when analyzing method body return type

### Steps To Reproduce

```c#
class Foo
{
        private static object Test(int arg)
        {
            if (arg > 4)
                return new List<string>();

            return arg switch
            {
                0 => new List<int>(),
                1 => new HashSet<int>(),
                _ => new Dictionary<long, int>(),
            };
        }
}
```

### Expected behavior

No CA1859 reported

### Actual behavior

```
CA1859: Change return type of method 'Test' from 'object' to 'System.Collections.Generic.List<string>' for improved performance
```

**Comment by stephentoub** [comment]
cc: @geeknoid

**Comment by geeknoid** [comment]
I don't quite understand.

The method as written above is returning a List, a HashSet, and a Dictionary. So recommending to upgrade the return type to List<string> would not be correct.

So what am I missing?

**Comment by stephentoub** [comment]
> So recommending to upgrade the return type to List would not be correct.

Right, but apparently the analyzer _is_ recommending that, ac ording to Marek.

**Comment by geeknoid** [comment]
Duh, I interpreted in reverse direction :-)
